media/js/base/fxa-utm-referral.js

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function () {
+
+    var UtmUrl = {};
+
+    UtmUrl.getAttributionData = function () {
+        var params = new window._SearchParams().utmParams();
+        var finalParams = {};
+
+        // utm_source
+        if (params.hasOwnProperty('utm_source') && (/^[\w/.%-]+$/).test(params['utm_source'])) {
+            // this value is already URI encoded, so we need to decode here as
+            // it will get re-encoded in _SearchParams.objectToQueryString
+            finalParams['source'] = decodeURIComponent(params['utm_source']);
```

Comment:
```suggestion
            finalParams['utm_source'] = decodeURIComponent(params['utm_source']);
```

You want to replace the `utm_source` parameter value, not change it to `source`.